app/models/post.rb

```diff
@@ -3,6 +3,7 @@ class Post < ApplicationRecord
   has_many :comments
   belongs_to :author, class_name: 'User', foreign_key: 'user_id'
 
```

Comment:
- [ ] I am not able to create a post, the issue is probably coming from here if a post `belongs_to: author` then foreign_key should be `author_id` not `user_id` and remember this will affect your columns in your table please address that
![post1](https://user-images.githubusercontent.com/44745855/160921473-78117c7c-09b4-4a44-9927-3334434814fa.png)

Reply:
Hi @DeliceLydia,

Thank you for the feedback, I actually intended naming the the column "user_id" in the beginning. The problem was in the `after_save` calling the `set_user_posts_counter` method, where I was using `user` instance variable instead of `author`.